src/app/app.element.ts

```diff
@@ -1,104 +1,102 @@
 import '@umbraco-ui/uui-css/dist/uui-css.css';
-import './core/css/custom-properties.css';
-
 import 'element-internals-polyfill';
 
-import './core/router/router-slot.element';
-import './core/router/variant-router-slot.element';
-import './core/context-provider/context-provider.element';
+import '../core/router/router-slot.element';
+import '../core/router/variant-router-slot.element';
+import '../core/context-provider/context-provider.element';
 
 import { UUIIconRegistryEssential } from '@umbraco-ui/uui';
 import { css, html } from 'lit';
 import { customElement, property } from 'lit/decorators.js';
 
-import { UmbAuthFlow } from './core/auth/auth-flow';
-import { UmbIconStore } from './core/stores/icon/icon.store';
-import type { UmbErrorElement } from './error/error.element';
+import { UmbAuthFlow } from '../core/auth/auth-flow';
+import { UmbIconStore } from '../core/stores/icon/icon.store';
+import type { UmbErrorElement } from '../error/error.element';
 import { UMB_APP, UmbAppContext } from './app.context';
+import { UmbAppConfig } from './app-config.interface';
 import type { Guard, UmbRoute } from '@umbraco-cms/backoffice/router';
 import { pathWithoutBasePath } from '@umbraco-cms/backoffice/router';
 import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
-import { UMB_SERVER_URL, tryExecute } from '@umbraco-cms/backoffice/resources';
+import { tryExecute } from '@umbraco-cms/backoffice/resources';
 import { OpenAPI, RuntimeLevelModel, ServerResource } from '@umbraco-cms/backoffice/backend-api';
 import { contextData, umbDebugContextEventType } from '@umbraco-cms/backoffice/context-api';
 
 @customElement('umb-app')
 export class UmbAppElement extends UmbLitElement {
-	/**
-	 * The base URL of the configured Umbraco server.
-	 *
-	 * @attr
-	 * @remarks This is the base URL of the Umbraco server, not the base URL of the backoffice.
-	 */
-	@property({ type: String })
-	private serverUrl = import.meta.env.VITE_UMBRACO_API_URL ?? '';
-
-	/**
-	 * The base path of the backoffice.
-	 *
-	 * @attr
-	 */
-	@property({ type: String })
-	// TODO: get from server config
-	private backofficePath = import.meta.env.DEV ? '/' : '/umbraco';
+	// for some reason it is not possible to use private fields for objects.
+	_config: UmbAppConfig = {
+		serverUrl: window.location.origin,
+		backofficePath: '/umbraco',
+		bypassAuth: false,
+	};
+
+	@property({ type: Object, attribute: false })
+	set config(value: UmbAppConfig) {
```

Comment:
Previously I kept these as separate properties with simple values so that they could be set from the backend Razor template in Light DOM individually, e.g.
```
<umb-app backoffice-path="/umbraco"></umb-app>
```

But I don't think I can do that after this change. Is it problematic to keep them as separate properties?

Reply:
Hm, that's a good point. I made a combined config because it is easier to know when to initialize the app. If we have more properties I had to check across. 

Just thinking about it, maybe it could be solved with Lis updated callback? 🤔

Reply:
Can't we just assume that any properties are initialized on the connected callback? I mean, either they are set through an environment variable or hardcoded into an attribute on the element, right?

Reply:
I have updated to use the connectedCallback 👍 . Can you have a second look?

Reply:
LGTM